feat(index.js): Address #8 (File rename) and #9 (save notes)

derikb opened the pull request:
Rename files to something meaningful and option to save post text as a txt file

Updated pizza-guy version too.
#8 #9

derikb committed:
feat(index.js): Address #8 (File rename) and #9 (save notes)

Rename files to something meaningful and option to save post text as a txt file

#8 #9

derikb commented:
@andresdavid90 This is the code that I've tested and used on my Tumblr using the offset param.

andrscrrn commented on README.md:
```diff
@@ -11,6 +11,8 @@ Simple node module for downloading your precious [Tumblr](https://tumblr.com) li
 A few things for having in mind while using it:
 * This will save ***just image files*** and not any other kind of content.
 * The number of `postsToLoad` it likely ***NOT*** going to be the same as the number of images that you will get. Posts can have more than one image or not having any image ***AT ALL***.
+* The maximum value for postsOffset is 1000.
```
Can you please add [this](https://www.tumblr.com/docs/en/api/v2#blog-likes) url here to let the users know that this maximum value is coming from the Tumblr API?

andrscrrn commented on src/index.js:
```diff
@@ -60,6 +65,17 @@ const downloadImages = () => {
+ * @param {String} filename a potential filename
+ * @returns {String} filename with characters stripped out
+ */
+const cleanFileName = function (filename) {
```
Do we really need this? I have never seen any problematic characters coming from the Tumblr API. I think this is a nice and smart function but not really sure if we really need it in this module. Can you please explain a little further about what was happening?

derikb replied:
This is just in case the tumblr user/blog name has problematic characters in it. Since the blog name will be the same throughout, I could make it more efficient by only running it once on the blog name (rather than how I have it running for every post).

andrscrrn replied:
String transformations shouldn't be too bad in terms of performance, even if you are doing it over and over. I'm pretty sure that downloading and saving on disk is a lot more expensive than that.

And about the reason of having this, it makes sense.

andrscrrn commented on src/index.js:
```diff
@@ -115,7 +138,11 @@ const getLikedPosts = (timestamp) => {
             const currentPost = likedPosts[i];
             if (Array.isArray(currentPost.photos)) {
-              currentPost.photos.forEach(pushToArray);
+              const filebase = cleanFileName(`${currentPost.blog_name}_${currentPost.timestamp}`);
```
What do you think if we create some kind of optional formats for the file names where this can be just one of them? The problem with this change is that at least me, I'm relying a lot on the current names and if I merge this change, I'm going to have a hard time trying to see which files were already there. Do you know what I mean?

derikb replied:
Sure, that's fine:
-Original image name
-Blogname_timestamp
-timestamp (?)

Not sure what else would be meaningful.

andrscrrn replied:
We can start with that for now I think. We can have the first option as the default.

derikb replied:
Thinking about it more, timestamp by itself would not necessarily be unique, so that's probably a bad idea.
How I about I just set it up as:

"If you want notes, then the files are named to {blogname}_{timestamp}{image number if applicable}"

that only adds one param then. Keeps things simpler.

andrscrrn replied:
Agree. Let's add that as a new property for the config object called `attribution`.

People would need to:

``` javascript
tumblrLksDownldr.setGlobalParams(
  {
    url: 'yourblog.tumblr.com',
    ...
    attribution: true,
    ...
  }
);
tumblrLksDownldr.getLikedPosts();
```

What do you think?

derikb replied:
Sounds good and easy enough. I'll just adjust my code a bit on here and then for the cli tool.

andrscrrn commented on src/index.js:
```diff
@@ -115,7 +138,11 @@ const getLikedPosts = (timestamp) => {
+              const filebase = cleanFileName(`${currentPost.blog_name}_${currentPost.timestamp}`);
+              currentPost.photos.forEach(pushToArray.bind(null, filebase));
+              if (includeNotes === true && currentPost.hasOwnProperty('caption')) {
+                saveNote(filebase, currentPost.caption);
```
I didn't know that this could be an important thing to others. Great that you want to add this. 🙃

derikb committed:
merge upstream

derikb committed:
feat(index): link filename change to attribution param

change note param to attribution and set filename change based on attribution

#8 #9

derikb commented:
Ok, updated the pull request for what we discussed above. New param is "attribution" defaults to false. If set to true then files are renamed "{blogname}_{timestamp}" and if the post has a caption it's converted to text and saved as a .txt file.

andresdavid90 commented:
@derikb Sorry for the delay... I just checked this again and it seems to be working just fine! Merged.
